pkg/cmd/version/version.go

```diff
@@ -18,21 +18,109 @@ package version
 
 import (
 	"fmt"
+	"time"
 
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
 
+	v1 "github.com/heptio/ark/pkg/apis/ark/v1"
+	"github.com/heptio/ark/pkg/builder"
 	"github.com/heptio/ark/pkg/buildinfo"
+	"github.com/heptio/ark/pkg/client"
+	"github.com/heptio/ark/pkg/cmd"
+	arkv1client "github.com/heptio/ark/pkg/generated/clientset/versioned/typed/ark/v1"
 )
 
-func NewCommand() *cobra.Command {
+func NewCommand(f client.Factory) *cobra.Command {
+	timeout := time.Minute
+	clientOnly := false
+
 	c := &cobra.Command{
 		Use:   "version",
 		Short: "Print the ark version and associated image",
-		Run: func(cmd *cobra.Command, args []string) {
-			fmt.Printf("Version: %s\n", buildinfo.Version)
-			fmt.Printf("Git commit: %s\n", buildinfo.FormattedGitSHA())
+		Run: func(c *cobra.Command, args []string) {
+			fmt.Println("Client:")
```

Comment:
I know we haven't done this much (if any) to date, but is this worth splitting out into its own function so we can unit test it?

Reply:
OK, refactored & added a unit test for what I think is the important part. PTAL.